client/templates/userprofile/settings.js

```diff
@@ -130,19 +176,42 @@ Template.profileSettings.events({
             if (Meteor.userId()) {
             Accounts.changePassword(oldpassword,password, function(error) {
                 if(error) {
-                    console.log("Something went wrong! " + error);
+                    if(error.reason === "Match failed")
+                    {
+                        Session.set("perrors", "Your old password is empty!");
+                    }
+                    else if(error.reason === "Incorrect password")
+                    {
+                        Session.set("perrors", "Your old Password is incorrect!");
+                    }
+                    else
+                    {
+                        Session.set("perrors", error.reason);
+                    }
+
 
                 }
-                else {
+                else 
+                {
                     Session.set("perrors", "");
                     Session.set("passwordsuccess","Password changed!");
                 }
             });
             }
         }
-        else {
+        else 
+        {
             Session.set("perrors", "Your passwords do not match.")
         }
+        setTimeout(function() {
```

Comment:
sessionTimeout may cause bugs and unnecessary delays. Use reactive data sources to update UI. Use SimpleSchema and Autoform for field validation.